flows/common/common.go

```diff
@@ -29,6 +30,73 @@ type matchFunc func([]gopacket.SerializableLayer, gopacket.Packet) bool
 // hdrs.
 type bpfFunc func([]gopacket.SerializableLayer) (string, error)
 
+// PortCreator is an interface for creating ports.
+type PortCreator interface {
+	CreatePort(name string) (Port, error)
+}
+
+// Port is interface for reading and writing to a port.
+type Port interface {
+	gopacket.PacketDataSource
+	// WritePacketData writes the frame to the port.
+	WritePacketData(data []byte) error
+	// Close closes the port.
+	Close()
+	// LinkType the layer which packet are processed.
+	LinkType() layers.LinkType
+	// SetSetBPFFilter sets the filter.
+	SetBPFFilter(string) error
+}
+
+// pcapPort contains both an inactive and active handle.
+type pcapPort struct {
+	*pcap.Handle
+	ih *pcap.InactiveHandle
+}
+
+// Close cleans up the inactive handle and closes the active one.
+func (p *pcapPort) Close() {
+	p.ih.CleanUp()
+	p.Handle.Close()
+}
+
+// pcapCreator implements the PortCreator interface for pcap handles.
+type pcapCreator struct {
+}
+
+var _ PortCreator = &pcapCreator{}
+
+// CreatePort creates a new port using a pcap handle.
+func (p *pcapCreator) CreatePort(name string) (Port, error) {
```

Comment:
Should we call this something like `CreateListener` or even `CreateCapture` -- since it's not actually creating the port (e.g., creating a new netdev interface or virtual one)?

Reply:
agreed, though i prefer Handle because since you can read and write to it. (to me listener/capture imply read only)

Reply:
Makes sense to me.